sdk/tables/azure-data-tables/CMakeLists.txt

```diff
@@ -0,0 +1,100 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../../../cmake-modules")
+include(AzureVcpkg)
+az_vcpkg_integrate()
+
+cmake_minimum_required (VERSION 3.13)
+project(azure-data-tables LANGUAGES CXX)
+
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED True)
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+
+option(FETCH_SOURCE_DEPS "build source dependencies" OFF)
+
+include(AzureVersion)
+include(AzureCodeCoverage)
+include(AzureTransportAdapters)
+include(AzureDoxygen)
+include(AzureGlobalCompileOptions)
+include(AzureConfigRTTI)
+include(AzureBuildTargetForCI)
+include(CreateMapFile)
+
+if(FETCH_SOURCE_DEPS)
+    set(AZ_ALL_LIBRARIES ON)
+    include(FolderList)
+    SetCompileOptions(DATA_TABLES)
+    GetFolderList(DATA_TABLES)
+    foreach(oneFolder IN LISTS BUILD_FOLDERS)
+        message("add folder ${oneFolder}")
+        add_subdirectory(${oneFolder} EXCLUDE_FROM_ALL)
+    endforeach()
+elseif(NOT AZ_ALL_LIBRARIES)
+  find_package(azure-storage-common-cpp CONFIG QUIET)
+  if(NOT azure-storage-common-cpp_FOUND)
+    find_package(azure-storage-common-cpp REQUIRED)
+  endif()
+endif()
+
+set(
+  AZURE_DATA_TABLES_HEADER
+    inc/azure/data/tables/dll_import_export.hpp
+    inc/azure/data/tables/rest_client.hpp
```

Comment:
Were the rest_client.hpp and rest_client.cpp files code-generated? We reserve those names, currently, for that.

For hand-written files, use more meaningful (and distinct) names.

Reply:
keyvault , queues have the same files and they were not auto generates

Reply:
Queues **is** code generated.
https://github.com/Azure/azure-sdk-for-cpp/blob/e328665588d2998fea198972af90d5108e64a968/sdk/storage/azure-storage-queues/src/rest_client.cpp#L4-L6

Where in keyvault do we have a `rest_client.cpp/.hpp` file? We'd want to fix that too (but I don't see it).

If you are referring to this, we should update it (it hasn't GA'd yet so thankfully we can), as we have to **reserve** the internal/generated "protocol-layer" code and the public facing included headers. The other GA'd keyvault SDKs don't have this pattern, which is good.
https://github.com/Azure/azure-sdk-for-cpp/blob/e328665588d2998fea198972af90d5108e64a968/sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/rest_client_models.hpp

Please rename the files based on the content of the file. As a strawman, you could consider `tables_client.cpp`, but I haven't looked further into what other contents are in there.

Reply:
done